// village/dist/src/common/auth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.warnUnauthenticated = exports.setAuthHeader = exports.getFile = exports.getWorkspaces = exports.getTokens = void 0;
const tslib_1 = require("tslib");
const _config_1 = require("@config");
const fs_1 = tslib_1.__importDefault(require("fs"));
const villageClient_1 = require("@common/villageClient");
const getTokens = async ({ debug } = { debug: false }) => {
    return (0, exports.getFile)({ filePath: _config_1.TOKENS_FILE, debug });
};
exports.getTokens = getTokens;
const getWorkspaces = async ({ debug } = { debug: false }) => {
    return (0, exports.getFile)({ filePath: _config_1.WORKSPACES_FILE, debug });
};
exports.getWorkspaces = getWorkspaces;
const getFile = async ({ filePath, debug } = {
    filePath: '',
    debug: false,
}) => {
    try {
        const data = await fs_1.default.readFileSync(filePath, 'utf8');
        // Error handling?
        return JSON.parse(data.toString());
    }
    catch (err) {
        debug && console.log(err);
        return null;
    }
};
exports.getFile = getFile;
const setAuthHeader = async ({ debug } = { debug: false }) => {
    const tokens = await (0, exports.getTokens)({ debug });
    if (tokens === null) {
        console.error('Account not found. Please log in by running `village login`.');
        return false;
    }
    const { access_token } = tokens;
    villageClient_1.villageClient.request.config.HEADERS = {
        Authorization: `Bearer ${access_token}`,
    };
    return true;
};
exports.setAuthHeader = setAuthHeader;
const warnUnauthenticated = async (err) => {
    if (err.status === 401) {
        console.error('Authentication failed. Please refresh by running `village logout` and then `village login`.');
    }
    throw err;
};
exports.warnUnauthenticated = warnUnauthenticated;
