// village/dist/src/commands/init.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createScript = void 0;
const tslib_1 = require("tslib");
const inquirer_1 = tslib_1.__importDefault(require("inquirer"));
const api_1 = require("../../api");
const fs_1 = tslib_1.__importDefault(require("fs"));
const js_yaml_1 = tslib_1.__importDefault(require("js-yaml"));
const villageClient_1 = require("@common/villageClient");
const configFile_1 = require("@common/configFile");
const auth_1 = require("@common/auth");
// ensure this is in sync with frontend/src/pages/NewScript.tsx
const engineVersions = {
    [api_1.Engine.NODE]: ['16', '17', '18'],
    [api_1.Engine.PYTHON]: ['3.8', '3.9', '3.10'],
};
const init = (debug) => {
    const questions = [
        {
            type: 'input',
            name: 'name',
            message: 'Script name:',
        },
        {
            type: 'input',
            name: 'id',
            message: `Script id (leave blank to autogenerate):`,
        },
        // TODO: load workspace IDs / names from server
        {
            type: 'input',
            name: 'workspace_id',
            message: `Workspace ID:`,
        },
        {
            type: 'input',
            name: 'description',
            message: `Script description:`,
        },
        {
            type: 'list',
            name: 'engine',
            message: 'Script engine:',
            choices: Object.values(api_1.Engine),
        },
    ];
    inquirer_1.default
        .prompt(questions)
        .then(({ name, id, description, engine, workspace_id, }) => {
        inquirer_1.default
            .prompt([
            {
                type: 'list',
                name: 'engineVersion',
                message: 'Engine version:',
                choices: engineVersions[engine],
            },
        ])
            .then(({ engineVersion }) => {
            villageClient_1.villageClient.scripts
                .createScript({
                id,
                name,
                description,
                engine,
                engine_version: engineVersion,
                workspace_id,
            })
                .then((script) => {
                (0, configFile_1.initializeScriptYaml)(Object.assign(Object.assign({}, script), { params: [], workspace: workspace_id }));
            })
                .catch(auth_1.warnUnauthenticated)
                .catch((err) => {
                debug && console.error(err);
            });
        })
            .catch((err) => {
            debug && console.error(err);
        });
    })
        .catch((err) => {
        if (err.status === 400) {
            console.error(err.body.detail);
            console.log('Run with --debug to see full error');
        }
        debug && console.error(err);
    });
};
const initializeScriptFrom = ({ from, debug, }) => {
    // check that file exists
    if (!fs_1.default.existsSync(from)) {
        console.error(`Config file ${from} does not exist`);
        return;
    }
    // read config file
    const configContents = fs_1.default.readFileSync(from, 'utf8');
    const config = js_yaml_1.default.load(configContents);
    villageClient_1.villageClient.scripts
        .createScript({
        id: config.id,
        name: config.name,
        engine: config.engine,
        engine_version: config.engine_version,
        workspace_id: config.workspace,
    })
        .catch(auth_1.warnUnauthenticated)
        .catch((err) => {
        if (err.status === 400) {
            console.error(err.body.detail);
            console.log('Run with --debug to see full error');
        }
        debug && console.error(err);
    });
};
const initializeExistingScript = ({ debug }) => {
    // if village.yaml already exists, don't overwrite it
    if (fs_1.default.existsSync('./village.yaml')) {
        inquirer_1.default
            .prompt([
            {
                type: 'confirm',
                name: 'confirm',
                message: 'village.yaml already exists. Overwrite?',
                default: false,
            },
        ])
            .then((answers) => {
            if (answers.confirm) {
                init(debug);
            }
        });
    }
    else {
        init(debug);
    }
};
const createScript = (program) => {
    program
        .command('init')
        .description('Init a new script')
        .option('--from <path>', 'Create a script from an existing config file (village.yaml).')
        .action(async function () {
        const debug = this.opts().debug;
        const authed = await (0, auth_1.setAuthHeader)({ debug });
        if (!authed) {
            return;
        }
        const from = this.opts().from;
        // if initializing from a config file, read it and initialize script
        if (from) {
            initializeScriptFrom({ from, debug });
        }
        else {
            initializeExistingScript({ debug });
        }
    });
};
exports.createScript = createScript;
