// village/dist/src/commands/login.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.login = void 0;
const tslib_1 = require("tslib");
const axios_1 = tslib_1.__importDefault(require("axios"));
const chalk_1 = tslib_1.__importDefault(require("chalk"));
const villageClient_1 = require("@common/villageClient");
const fs_1 = tslib_1.__importDefault(require("fs"));
const qs_1 = tslib_1.__importDefault(require("qs"));
const _config_1 = require("@config");
const auth_1 = require("@common/auth");
const path_1 = require("path");
const userinfo_1 = require("./userinfo");
// see https://stackoverflow.com/questions/16316330/how-to-write-file-if-parent-folder-doesnt-exist
const writeFile = async (path, contents, callback) => {
    await fs_1.default.mkdir((0, path_1.dirname)(path), { recursive: true }, function (err) {
        if (err)
            return callback(err);
        fs_1.default.writeFile(path, contents, callback);
    });
};
const getRequestAccessTokenOptions = (deviceCode) => {
    return {
        method: 'POST',
        url: `${_config_1.AUTH0_DOMAIN}/oauth/token`,
        headers: { 'content-type': 'application/x-www-form-urlencoded' },
        data: new URLSearchParams({
            grant_type: 'urn:ietf:params:oauth:grant-type:device_code',
            device_code: deviceCode,
            client_id: _config_1.AUTH0_CLIENT_ID,
            audience: _config_1.AUTH0_AUDIENCE,
        }),
    };
};
const pollForAccessToken = (deviceCode) => {
    return new Promise((resolve, reject) => {
        const interval = setInterval(() => {
            axios_1.default
                .request(getRequestAccessTokenOptions(deviceCode))
                .then((response) => {
                if (response.data.access_token) {
                    clearInterval(interval);
                    resolve(response.data);
                }
            })
                .catch((err) => {
                if (err.response.data.error === 'authorization_pending') {
                    // TODO: do something here
                }
                else if (err.response.data.error === 'slow_down') {
                    // TODO: do something here
                }
                else if (err.response.data.error === 'expired_token') {
                    console.error('Authorization expired. Please try again.');
                    clearInterval(interval);
                    reject(err);
                }
                else if (err.response.data.error === 'access_denied') {
                    console.error('Authorization denied. Please try again.');
                    clearInterval(interval);
                    reject(err);
                }
            });
        }, 5000);
    });
};
const initializeUser = async (user, accessToken) => {
    if (!user.sub)
        return;
    let workspaces;
    try {
        villageClient_1.villageClient.request.config.HEADERS = {
            Authorization: `Bearer ${accessToken}`,
        };
        const res = await villageClient_1.villageClient.user.getCurrentUser();
        workspaces = res.workspaces;
    }
    catch (error) {
        if (error.message === 'Not Found') {
            const res = await villageClient_1.villageClient.user.createUser(user.sub);
            workspaces = res.workspaces;
        }
    }
    await writeFile(_config_1.WORKSPACES_FILE, JSON.stringify(workspaces), () => {
        return;
    });
};
const login = (program) => {
    program
        .command('login')
        .description('Login to Village')
        .action(async function () {
        const debug = this.opts().debug;
        const tokens = await (0, auth_1.getTokens)({ debug });
        if (tokens !== null) {
            console.log("You're logged in! Run `village logout` to log out first.");
            return;
        }
        try {
            const res = await axios_1.default.request({
                method: 'POST',
                url: `${_config_1.AUTH0_DOMAIN}/oauth/device/code`,
                headers: {
                    'content-type': 'application/x-www-form-urlencoded',
                },
                data: qs_1.default.stringify({
                    client_id: _config_1.AUTH0_CLIENT_ID,
                    scope: 'openid email profile offline_access',
                    audience: _config_1.AUTH0_AUDIENCE,
                }),
            });
            const { verification_uri } = res.data;
            console.log('Complete authorization at');
            console.log(chalk_1.default.blue.underline.bgWhite(verification_uri), '\n');
            console.log('Your auth code is');
            console.log(chalk_1.default.blue.bgWhite(res.data.user_code), '\n');
            const data = await pollForAccessToken(res.data.device_code);
            debug && console.log(data);
            console.log('tokens', data);
            await writeFile(_config_1.TOKENS_FILE, JSON.stringify(data), () => {
                console.log('Successfully logged in!');
            });
            const { data: userInfo, errors } = await (0, userinfo_1.getAuth0UserInfo)(data);
            debug && console.error(errors);
            await initializeUser(userInfo, data.access_token);
        }
        catch (error) {
            debug && console.error(error);
        }
    });
};
exports.login = login;
