// village/dist/src/commands/setup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupScript = void 0;
const auth_1 = require("@common/auth");
const configFile_1 = require("@common/configFile");
const villageClient_1 = require("@common/villageClient");
const setupScript = (program) => {
    program
        .command('setup')
        .description('Set up a new script')
        .argument('<id>', 'ID of the script to set up.')
        .action(async function () {
        const debug = this.opts().debug;
        const authed = await (0, auth_1.setAuthHeader)({ debug });
        if (!authed) {
            return;
        }
        const id = program.args[1];
        // if village.yaml exists, ask before overwriting
        villageClient_1.villageClient.scripts
            .getScript(id)
            .then((s) => {
            (0, configFile_1.initializeScriptYaml)({
                id: s.id,
                name: s.name,
                workspace: s.workspace_id,
                engine: s.engine,
                engine_version: s.engine_version,
                params: [],
            });
        })
            .catch(auth_1.warnUnauthenticated)
            .catch((err) => {
            if (err.status === 404) {
                console.log(`Script ${id} not found.`);
            }
            debug && console.error(err);
        });
    });
};
exports.setupScript = setupScript;
