// village/dist/src/commands/workspaces.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.chooseWorkspace = void 0;
const tslib_1 = require("tslib");
const auth_1 = require("@common/auth");
const villageClient_1 = require("@common/villageClient");
const prompts_1 = tslib_1.__importDefault(require("prompts"));
const chooseWorkspace = (program) => {
    program
        .command('workspace')
        .description('Choose workspace')
        .action(async function () {
        const debug = this.opts().debug;
        const tokens = await (0, auth_1.getTokens)({ debug });
        const { access_token } = tokens;
        try {
            villageClient_1.villageClient.request.config.HEADERS = {
                Authorization: `Bearer ${access_token}`,
            };
            let workspacesForUser;
            try {
                workspacesForUser =
                    await villageClient_1.villageClient.workspace.getWorkspaceByUser();
            }
            catch (error) {
                await (0, auth_1.warnUnauthenticated)(error);
            }
            const selection = await (0, prompts_1.default)([
                {
                    type: 'select',
                    name: 'value',
                    message: 'Choose workspace',
                    choices: workspacesForUser.map((workspaceForUser) => {
                        return {
                            title: workspaceForUser.workspace.name,
                            value: workspaceForUser.workspace_id,
                        };
                    }),
                },
            ]);
            const setDefaultResponse = await villageClient_1.villageClient.workspace.setDefaultWorkspace(selection.value);
            setDefaultResponse && console.log('Default workspace updated');
        }
        catch (error) {
            console.error(error);
        }
    });
};
exports.chooseWorkspace = chooseWorkspace;
