// village/dist/api/services/BuildsService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BuildsService = void 0;
class BuildsService {
    constructor(httpRequest) {
        this.httpRequest = httpRequest;
    }
    /**
     * Get Build
     * Get a build.
     * @param buildId
     * @returns any Successful Response
     * @throws ApiError
     */
    getBuild(buildId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/build/get',
            query: {
                'build_id': buildId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * List Builds
     * List all builds.
     * @returns any Successful Response
     * @throws ApiError
     */
    listBuilds() {
        return this.httpRequest.request({
            method: 'GET',
            url: '/build/list',
        });
    }
    /**
     * Get Build Runs
     * Get all runs for a build.
     * @param buildId
     * @returns any Successful Response
     * @throws ApiError
     */
    getBuildRuns(buildId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/build/runs',
            query: {
                'build_id': buildId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Delete Build
     * Delete a build.
     * @param buildId
     * @returns Build Successful Response
     * @throws ApiError
     */
    deleteBuild(buildId) {
        return this.httpRequest.request({
            method: 'DELETE',
            url: '/build/delete',
            query: {
                'build_id': buildId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
}
exports.BuildsService = BuildsService;
