// village/dist/api/services/RunsService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RunsService = void 0;
class RunsService {
    constructor(httpRequest) {
        this.httpRequest = httpRequest;
    }
    /**
     * List Runs
     * List all runs.
     * @returns RunWithBuilds Successful Response
     * @throws ApiError
     */
    listRuns() {
        return this.httpRequest.request({
            method: 'GET',
            url: '/run/list',
        });
    }
    /**
     * Get Run
     * Get a run.
     * @param runId
     * @returns Run Successful Response
     * @throws ApiError
     */
    getRun(runId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/run/get',
            query: {
                'run_id': runId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Delete Run
     * Delete a run.
     * @param runId
     * @returns Run Successful Response
     * @throws ApiError
     */
    deleteRun(runId) {
        return this.httpRequest.request({
            method: 'DELETE',
            url: '/run/delete',
            query: {
                'run_id': runId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
}
exports.RunsService = RunsService;
