// village/dist/api/services/SchedulesService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchedulesService = void 0;
class SchedulesService {
    constructor(httpRequest) {
        this.httpRequest = httpRequest;
    }
    /**
     * Create Schedule
     * Add a script schedule to the scheduler
     * @param requestBody
     * @returns Schedule Successful Response
     * @throws ApiError
     */
    createSchedule(requestBody) {
        return this.httpRequest.request({
            method: 'POST',
            url: '/schedule/create',
            body: requestBody,
            mediaType: 'application/json',
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Update Schedule
     * @param requestBody
     * @returns Schedule Successful Response
     * @throws ApiError
     */
    updateSchedule(requestBody) {
        return this.httpRequest.request({
            method: 'POST',
            url: '/schedule/update',
            body: requestBody,
            mediaType: 'application/json',
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Run Schedule
     * Execute a script based on its schedule entry
     * @param scheduleId
     * @returns Run Successful Response
     * @throws ApiError
     */
    runSchedule(scheduleId) {
        return this.httpRequest.request({
            method: 'POST',
            url: '/schedule/run',
            query: {
                'schedule_id': scheduleId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * List Schedules
     * List all schedules
     * @returns ScheduleWithScript Successful Response
     * @throws ApiError
     */
    listSchedules() {
        return this.httpRequest.request({
            method: 'GET',
            url: '/schedule/list',
        });
    }
    /**
     * Delete Schedule
     * Delete a script schedule from the scheduler
     * @param scheduleId
     * @returns Schedule Successful Response
     * @throws ApiError
     */
    deleteSchedule(scheduleId) {
        return this.httpRequest.request({
            method: 'DELETE',
            url: '/schedule/delete',
            query: {
                'schedule_id': scheduleId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
}
exports.SchedulesService = SchedulesService;
