// village/dist/api/services/ScriptsService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScriptsService = void 0;
class ScriptsService {
    constructor(httpRequest) {
        this.httpRequest = httpRequest;
    }
    /**
     * Propose Id
     * Propose a script ID based on the name.
     * @param name
     * @returns string Successful Response
     * @throws ApiError
     */
    proposeId(name) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/scripts/propose_id',
            query: {
                'name': name,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Check Id
     * Check if a script ID is available.
     * @param id
     * @returns boolean Successful Response
     * @throws ApiError
     */
    checkId(id) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/scripts/check_id',
            query: {
                'id': id,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Create Script
     * Initialize a script.
     * @param requestBody
     * @returns Script Successful Response
     * @throws ApiError
     */
    createScript(requestBody) {
        return this.httpRequest.request({
            method: 'POST',
            url: '/script/create',
            body: requestBody,
            mediaType: 'application/json',
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Update Script
     * Update script contents.
     * @param requestBody
     * @returns Script Successful Response
     * @throws ApiError
     */
    updateScript(requestBody) {
        return this.httpRequest.request({
            method: 'POST',
            url: '/script/update',
            body: requestBody,
            mediaType: 'application/json',
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Build Script
     * Build a script so that it can be run.
     * @param formData
     * @returns Build Successful Response
     * @throws ApiError
     */
    buildScript(formData) {
        return this.httpRequest.request({
            method: 'POST',
            url: '/script/build',
            formData: formData,
            mediaType: 'multipart/form-data',
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Run Script
     * Execute the latest build for a script.
     * @param requestBody
     * @returns Run Successful Response
     * @throws ApiError
     */
    runScript(requestBody) {
        return this.httpRequest.request({
            method: 'POST',
            url: '/script/run',
            body: requestBody,
            mediaType: 'application/json',
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * List Scripts
     * List all scripts.
     * @param workspaceId
     * @returns Script Successful Response
     * @throws ApiError
     */
    listScripts(workspaceId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/script/list',
            query: {
                'workspace_id': workspaceId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Get Script
     * Get a script.
     * @param scriptId
     * @returns ScriptWithBuild Successful Response
     * @throws ApiError
     */
    getScript(scriptId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/script/get',
            query: {
                'script_id': scriptId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Get Script Builds
     * Get all builds for a script.
     * @param scriptId
     * @returns Build Successful Response
     * @throws ApiError
     */
    getScriptBuilds(scriptId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/script/builds',
            query: {
                'script_id': scriptId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Get Script Schedules
     * Get all schedules for a script.
     * @param scriptId
     * @returns Schedule Successful Response
     * @throws ApiError
     */
    getScriptSchedules(scriptId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/script/schedules',
            query: {
                'script_id': scriptId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Get Script Runs
     * Get all runs for a script.
     * @param scriptId
     * @returns Run Successful Response
     * @throws ApiError
     */
    getScriptRuns(scriptId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/script/runs',
            query: {
                'script_id': scriptId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Delete Script
     * Delete a script.
     * @param scriptId
     * @returns Script Successful Response
     * @throws ApiError
     */
    deleteScript(scriptId) {
        return this.httpRequest.request({
            method: 'DELETE',
            url: '/script/delete',
            query: {
                'script_id': scriptId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
}
exports.ScriptsService = ScriptsService;
