// village/dist/api/services/WorkspaceService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceService = void 0;
class WorkspaceService {
    constructor(httpRequest) {
        this.httpRequest = httpRequest;
    }
    /**
     * Create Workspace
     * Create a workspace.
     * @param name
     * @returns Workspace Successful Response
     * @throws ApiError
     */
    createWorkspace(name) {
        return this.httpRequest.request({
            method: 'POST',
            url: '/workspace/create',
            query: {
                'name': name,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Get Workspace
     * Get a workspace.
     * @param workspaceId
     * @returns Workspace Successful Response
     * @throws ApiError
     */
    getWorkspace(workspaceId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/workspace/get',
            query: {
                'workspace_id': workspaceId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
    /**
     * Get Workspace By User
     * Get all workspaces by user.
     * @returns WorkspaceUsers Successful Response
     * @throws ApiError
     */
    getWorkspaceByUser() {
        return this.httpRequest.request({
            method: 'GET',
            url: '/workspace/get_by_user',
        });
    }
    /**
     * Set Default Workspace
     * Set default workspace.
     * @param workspaceId
     * @returns boolean Successful Response
     * @throws ApiError
     */
    setDefaultWorkspace(workspaceId) {
        return this.httpRequest.request({
            method: 'POST',
            url: '/workspace/set_default',
            query: {
                'workspace_id': workspaceId,
            },
            errors: {
                422: `Validation Error`,
            },
        });
    }
}
exports.WorkspaceService = WorkspaceService;
