// village/dist/api/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceService = exports.UserService = exports.ScriptsService = exports.SchedulesService = exports.RunsService = exports.BuildsService = exports.Role = exports.ParamType = exports.Engine = exports.OpenAPI = exports.CancelError = exports.CancelablePromise = exports.BaseHttpRequest = exports.ApiError = exports.VillageClient = void 0;
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */
var VillageClient_1 = require("./VillageClient");
Object.defineProperty(exports, "VillageClient", { enumerable: true, get: function () { return VillageClient_1.VillageClient; } });
var ApiError_1 = require("./core/ApiError");
Object.defineProperty(exports, "ApiError", { enumerable: true, get: function () { return ApiError_1.ApiError; } });
var BaseHttpRequest_1 = require("./core/BaseHttpRequest");
Object.defineProperty(exports, "BaseHttpRequest", { enumerable: true, get: function () { return BaseHttpRequest_1.BaseHttpRequest; } });
var CancelablePromise_1 = require("./core/CancelablePromise");
Object.defineProperty(exports, "CancelablePromise", { enumerable: true, get: function () { return CancelablePromise_1.CancelablePromise; } });
Object.defineProperty(exports, "CancelError", { enumerable: true, get: function () { return CancelablePromise_1.CancelError; } });
var OpenAPI_1 = require("./core/OpenAPI");
Object.defineProperty(exports, "OpenAPI", { enumerable: true, get: function () { return OpenAPI_1.OpenAPI; } });
var Engine_1 = require("./models/Engine");
Object.defineProperty(exports, "Engine", { enumerable: true, get: function () { return Engine_1.Engine; } });
var ParamType_1 = require("./models/ParamType");
Object.defineProperty(exports, "ParamType", { enumerable: true, get: function () { return ParamType_1.ParamType; } });
var Role_1 = require("./models/Role");
Object.defineProperty(exports, "Role", { enumerable: true, get: function () { return Role_1.Role; } });
var BuildsService_1 = require("./services/BuildsService");
Object.defineProperty(exports, "BuildsService", { enumerable: true, get: function () { return BuildsService_1.BuildsService; } });
var RunsService_1 = require("./services/RunsService");
Object.defineProperty(exports, "RunsService", { enumerable: true, get: function () { return RunsService_1.RunsService; } });
var SchedulesService_1 = require("./services/SchedulesService");
Object.defineProperty(exports, "SchedulesService", { enumerable: true, get: function () { return SchedulesService_1.SchedulesService; } });
var ScriptsService_1 = require("./services/ScriptsService");
Object.defineProperty(exports, "ScriptsService", { enumerable: true, get: function () { return ScriptsService_1.ScriptsService; } });
var UserService_1 = require("./services/UserService");
Object.defineProperty(exports, "UserService", { enumerable: true, get: function () { return UserService_1.UserService; } });
var WorkspaceService_1 = require("./services/WorkspaceService");
Object.defineProperty(exports, "WorkspaceService", { enumerable: true, get: function () { return WorkspaceService_1.WorkspaceService; } });
