// village/dist/api/VillageClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VillageClient = void 0;
const NodeHttpRequest_1 = require("./core/NodeHttpRequest");
const BuildsService_1 = require("./services/BuildsService");
const RunsService_1 = require("./services/RunsService");
const SchedulesService_1 = require("./services/SchedulesService");
const ScriptsService_1 = require("./services/ScriptsService");
const UserService_1 = require("./services/UserService");
const WorkspaceService_1 = require("./services/WorkspaceService");
class VillageClient {
    constructor(config, HttpRequest = NodeHttpRequest_1.NodeHttpRequest) {
        var _a, _b, _c, _d;
        this.request = new HttpRequest({
            BASE: (_a = config === null || config === void 0 ? void 0 : config.BASE) !== null && _a !== void 0 ? _a : '',
            VERSION: (_b = config === null || config === void 0 ? void 0 : config.VERSION) !== null && _b !== void 0 ? _b : '0.1.0',
            WITH_CREDENTIALS: (_c = config === null || config === void 0 ? void 0 : config.WITH_CREDENTIALS) !== null && _c !== void 0 ? _c : false,
            CREDENTIALS: (_d = config === null || config === void 0 ? void 0 : config.CREDENTIALS) !== null && _d !== void 0 ? _d : 'include',
            TOKEN: config === null || config === void 0 ? void 0 : config.TOKEN,
            USERNAME: config === null || config === void 0 ? void 0 : config.USERNAME,
            PASSWORD: config === null || config === void 0 ? void 0 : config.PASSWORD,
            HEADERS: config === null || config === void 0 ? void 0 : config.HEADERS,
            ENCODE_PATH: config === null || config === void 0 ? void 0 : config.ENCODE_PATH,
        });
        this.builds = new BuildsService_1.BuildsService(this.request);
        this.runs = new RunsService_1.RunsService(this.request);
        this.schedules = new SchedulesService_1.SchedulesService(this.request);
        this.scripts = new ScriptsService_1.ScriptsService(this.request);
        this.user = new UserService_1.UserService(this.request);
        this.workspace = new WorkspaceService_1.WorkspaceService(this.request);
    }
}
exports.VillageClient = VillageClient;
